oonidata/oonidata/s3feeder.py

```diff
@@ -0,0 +1,508 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Feeds reports from cans on public S3 bucke or local disk
+
+Explore bucket from CLI:
+AWS_PROFILE=ooni-data aws s3 ls s3://ooni-data/canned/2019-07-16/
+
+"""
+
+from datetime import date, timedelta, datetime
+from typing import Generator, Set, NamedTuple, Any, List
+from collections import namedtuple
+from pathlib import Path
+import itertools
+import logging
+import os
+import time
+import gzip
+import tarfile
+from multiprocessing import Pool
+
+import lz4.frame as lz4frame  # debdeps: python3-lz4
+import ujson
+
+# lz4frame appears faster than executing lz4cat: 2.4s vs 3.9s on a test file
+
+import boto3  # debdeps: python3-boto3
+from botocore import UNSIGNED as botoSigUNSIGNED
+from botocore.config import Config as botoConfig
+
+from .metrics import setup_metrics
+from .mytypes import MsmtTup  # msmt bytes, msmt dict, uid
+from .normalize import iter_yaml_msmt_normalized
+from .utils import trivial_id
+
+CAN_BUCKET_NAME = "ooni-data"
+MC_BUCKET_NAME = "ooni-data-eu-fra"
+MAX_PROCESS_COUNT = 24
+
+log = logging.getLogger("fastpath")
+metrics = setup_metrics(name="fastpath.s3feeder")
+
+
+def create_s3_client():
+    return boto3.client("s3", config=botoConfig(signature_version=botoSigUNSIGNED))
+
+s3 = create_s3_client()
```

Comment:
This part of the library will most likely not be exposed to end users. These methods are internal methods that we use inside of the data processing pipeline.

I agree with you that having globals is probably not ideal. What I did from the perspective of this PR was to minimise the amount of changes made to the existing codebase so that the likelyhood of breaking our pipeline is reduced.

Some refactoring is probably in order further down the line.

I also think that for end users we probably want to have a different, more user-friendly API, that abstracts away all of these alien concepts and just exposes a very simply "give me a generator of measurements for this search criteria".